Python/specialize.c

```diff
@@ -814,15 +819,47 @@ _Py_Specialize_LoadMethod(PyObject *owner, _Py_CODEUNIT *instr, PyObject *name,
     cache1->tp_version = owner_cls->tp_version_tag;
 
     if (kind != METHOD || descr == NULL) {
-#if SPECIALIZATION_STATS
         if (kind == GETSET_OVERRIDDEN &&
             PyObject_TypeCheck(owner, &PyBaseObject_Type)) {
-            // Maybe TODO: LOAD_METHOD_CLASS
-            SPECIALIZATION_FAIL(LOAD_METHOD, SPEC_FAIL_CLASS_METHOD);
-            goto fail;
+            PyTypeObject *cls = (PyTypeObject *)owner;
+            kind = analyze_descriptor(cls, name, &descr, 0);
+            // Store the version right away, in case it's modified halfway through.
+            cache1->tp_version = cls->tp_version_tag;
+            if (descr == NULL) {
+                SPECIALIZATION_FAIL(LOAD_METHOD, SPEC_FAIL_NOT_METHOD);
+                goto fail;
+            }
+            // Borrowed, only used for identity checks. Do not access the object.
+            _PyObjectCache *cache3 = &cache[-3].obj;
+
+            // Python class method
+            if (kind == METHOD) {
+                // borrowed, see below for why this is safe
+                cache2->obj = descr;
+                cache3->obj = owner;
+                *instr = _Py_MAKECODEUNIT(LOAD_METHOD_CLASS, _Py_OPARG(*instr));
+                goto success;
+            }
+#if SPECIALIZATION_STATS
+            // Builtin class method -- ie wrapped PyCFunction
+            else if (kind == NON_OVERRIDING && Py_IS_TYPE(descr, &PyClassMethodDescr_Type)) {
```

Comment:
I think that this is impossible. If `Py_IS_TYPE(descr, &PyClassMethodDescr_Type)` then `kind == METHOD`.

Reply:
I was shocked too, but anything with ``METH_CLASS | METH_FASTCALL`` results in that when searched via `_PyType_Lookup`.
E.g.
``dict.fromkeys([1])``

Reply:
Ah. I'd forgotten about `METH_CLASS`.